### PR TITLE
exception handling / it-academy / JC1-110-22 / Evgeniy Antonovich / Done tasks 1- 7

### PR DESCRIPTION
**Pull request by aevgeni**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task1Demo.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+public class Task1Demo {
+    public static void main(String[] args) {
+        Task1 testNullException = new Task1();
+
+        try {
+            testNullException.strToUpperCase();
+        } catch (Exception e) {
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task1Demo.java:9]: Best practice: catch a specific exception

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task1Demo.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+            System.out.println("Null pointer exception");
+            e.printStackTrace();
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task1Demo.java:11]: please add one sout

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task6.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+public class Task6 {
+    public void getOneOfThreeExceptions(int variant) throws ArithmeticException, ArrayIndexOutOfBoundsException, NullPointerException {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task6.java:4]: Violation for Specification of Tasks

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task7.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+import java.net.SocketException;
+
+public class Task7 {
+    public void getException(int variant) throws SocketException {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task7.java:6]: Violation for Specification of Tasks

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

**Comment by aevgeni** [comment]
Fixed all violation

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task6.java (hunk 3 above)
Comment (aevgeni) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task6.java]: fixed it

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task7.java (hunk 4 above)
Comment (aevgeni) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task7.java:6]: fixed it

**Review by rakovets — APPROVED** [review]

**Pull request closed without merging**